installers/pgo-installer/ansible/roles/pgo-installer/tasks/main.yml

```diff
@@ -0,0 +1,108 @@
+---
+- name: Set output directory fact
+  set_fact:
+    output_dir: "{{ ansible_env.HOME }}/.pgo/{{ pgo_operator_namespace }}/output"
+  tags: 
+    - always
+
+- name: Ensure output directory exists
+  file:
+    path: "{{ output_dir }}"
+    state: directory
+    mode: 0700
+  tags:
+    - always
+
+- include_vars: openshift.yml
+  when: use_oc | default(false) | bool
+  tags:
+    - always
+
+- name: Use kubectl or oc
+  set_fact:
+    kubectl_or_oc: "{{ openshift_oc_bin if openshift_oc_bin is defined else 'kubectl' }}"
+  tags:
+    - always
+
+- name: Setup Install Container
+  tags:
+    - always
+  block:
+  - name: Create Install Namespace
+    shell: "{{ kubectl_or_oc }} create namespace {{ pgo_installer_namespace }}"
+    ignore_errors: yes
+      
+  - name: Create PGO-Install Service Account
+    shell: "{{ kubectl_or_oc }} create serviceaccount {{ pgo_install_sa }} -n {{ pgo_installer_namespace }}"
+    ignore_errors: yes
+
+  - name: Create PGO-Installer Cluster Role Binding
+    shell: "{{ kubectl_or_oc }} create clusterrolebinding {{ pgo_install_crb }} -n {{ pgo_installer_namespace }} --clusterrole=cluster-admin --serviceaccount={{ pgo_installer_namespace }}:{{ pgo_install_sa }}"
+    ignore_errors: yes
+
+  - name: Create Inventory Config Map
+    shell: "{{ kubectl_or_oc }} create configmap pgo-install-inventory -n {{ pgo_installer_namespace }} --from-file=inventory={{ inventory_dir }}/inventory"
```

Comment:
It seems like this is the point where the caller gets to decide how the operator is configured. Where does `inventory_dir` come from?

Reply:
`(( inventory_dir ))` points to the directory that holds the inventory file that the user passed in. Seems like this could be updated to `inventory_file`.

Reply:
The same inventory is used by the person invoking the installer and the installer itself? If so, what switches from "install/create remote k8s objects" to "install/create in-cluster k8s objects"?

Reply:
>The same inventory is used by the person invoking the installer and the installer itself?

Yes, I'm definitely open to better ideas on how to handle this.

>If so, what switches from "install/create remote k8s objects" to "install/create in-cluster k8s objects"?

I'm not following.

Reply:
Perhaps I'll understand better with some docs. Right now, my assumption is that a person who is not inside the cluster will fill out their inventory file with `kubernetes_context='staging-or-so'` then run `ansible-playbook installers/pgo-installer/ansible/main.yml` to create the install container in the remote (staging) cluster. If so, when the install container runs, it will also have `kubernetes_context='staging-or-so'` (which isn't defined?)

Reply:
I do need to get some docs going and hopefully that would explain better. Right now, I'm not accounting for contexts for the installer. I assume that the user is logged into their kube/ocp cluster has the ability to run kubectl/oc commands to create resources in that cluster. For example, the user has already run `oc login` and could run `oc create ns` on the environment in which they are attempting to install.

`kubectl_or_oc` in this case would be which binary do they want the ansible roles for the ansible installer to use. I'm thinking about setting this up with the `pgo_installer_environment=kubernetes/oc` variable that @jkatz mentioned so that the "correct" binary is used.

Reply:
The user would fill out the inventory file in `$PGOROOT/installers/ansible` then use the following to run the install.
```
ansible-playbook -i $PGOROOT/installers/ansible/inventory --tags=install-container $PGOROOT/installers/pgo-installer/ansible/main.yml
```